src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -889,6 +889,48 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, submit: 'config', config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
```

Comment:
IMO we shouldn't be mocking the save here since this is being overly specific about the internals of the controller so it makes the test more brittle. The test should allow you to refactor the code in the controller without having to change the test and this doesn't allow that. I would say its better to have an expectation on the side effects of the ProjectConfigFile.save call i.e.
```ruby
it { expect(ProjectConfigFile.count).to eq(1) } ```
```
or something like that..